go-controller/pkg/ovn/controller/services/node_tracker.go

```diff
@@ -204,6 +207,7 @@ func (nt *nodeTracker) updateNode(node *v1.Node) {
 		grName,
 		ips,
 		hsn,
+		node.Labels,
```

Comment:
The `AddFunc` only checks for annotation and name changes, so we probably need to update it to account for the zone label change. Probably need a test in the service controller for this (as there are no node_tracker specific tests).
I would update this function doc as well.